src/pages/VenuePage/TemplateWrapper.tsx

```diff
@@ -145,19 +146,21 @@ const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
 
   // @debt remove backButton from Navbar
   return (
-    <RelatedVenuesProvider venueId={venue.id}>
-      <ReactionsProvider venueId={venue.id}>
-        <WithNavigationBar
-          fullscreen={fullscreen}
-          hasBackButton={hasBackButton}
-        >
-          <AnnouncementMessage message={venue.bannerMessage} />
-          {template}
-          <ChatSidebar venue={venue} />
-          <UserProfileModal venue={venue} />
-        </WithNavigationBar>
-      </ReactionsProvider>
-    </RelatedVenuesProvider>
+    <WorldUsersProvider venueId={venue.id}>
+      <RelatedVenuesProvider venueId={venue.id}>
+        <ReactionsProvider venueId={venue.id}>
+          <WithNavigationBar
+            fullscreen={fullscreen}
+            hasBackButton={hasBackButton}
+          >
+            <AnnouncementMessage message={venue.bannerMessage} />
```

Comment:
[NB] Context is cool and declarative but this is starting to look like 'wrapper hell', and that's exactly one of the benefits of react hooks, they can avoid this. Just noting. No change request required.

Reply:
What's your suggestion for the alternative? This is the canonical way that React uses context providers. Hooks can't help/change this in any way that I am aware of. At the 'usage site' we use the hooks for React's context API.

Reply:
We have redux, why not use it instead of context? It has the same benefits, avoiding props drilling and global access to the store.

Reply:
@denisdimitrov What does context have to do with props drilling..? You know that redux is built on top of React's context API's right...?

There are things that make sense in Redux. There are other things that are way harder if not impossible to do with it. Some choices not to use it so far have been boilerplate / time / existing patterns related. I want to move more to redux as our core pattern in future, but that doesn't make context a bad choice.

I'm still failing to see what your actual concerns/issues with it are..?